### PR TITLE
Multithreading and logging

### PR DESCRIPTION
**Pull request by atvaccaro**
This is in support of https://github.com/cal-itp/data-infra/pull/902; adds a configurable parallelization to executing the validation JAR. It also adds a better CLI entrypoint for local development and testing.

Example execution

```
python gtfs_rt_validator_api.py validate-gcs-bucket-many --param-csv=gs://gtfs-data-test/rt-processed/calitp_validation_params/2022-01-17.csv --threads=4 --limit=12
```

This PR also adds some pre-commit hooks, etc.

### CODE REVIEW

File: gtfs_rt_validator_api.py
Comment (lauriemerrell) [inline gtfs_rt_validator_api.py:286]: Do we still want this to be a print statement? Or a logging one? (Noticed that it seems like we're trying to move to logging statements instead...?)

File: rectangle_vehicles.py
Comment (lauriemerrell) [inline rectangle_vehicles.py:1]: Do we still need this file at all....? Wondering if it's superseded by all the processing in `data-infra`

File: gtfs_rt_validator_api.py
Comment (lauriemerrell) [inline gtfs_rt_validator_api.py:125]: Another maybe-comment spot?

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
Overall looks good, just a few small comments, I think more commenting could be helpful since this is introducing a bunch of new patterns that don't exist elsewhere in our codebase (...yet?) around the multiprocessing and the use of Typer.

**Comment by lauriemerrell** [comment]
Oh whoops, didn't realize this was still in Draft! My bad. @atvaccaro you can ignore the above until it's ready 😅

File: gtfs_rt_validator_api.py
Comment (atvaccaro) [reply gtfs_rt_validator_api.py]: yep! changed them all

File: rectangle_vehicles.py
Comment (atvaccaro) [reply rectangle_vehicles.py:1]: Probably not, I will mark as deprecated/unneeded without deleting to keep this PR smaller

File: gtfs_rt_validator_api.py
Comment (lauriemerrell) [inline gtfs_rt_validator_api.py:164]: Is this maybe more of a debug-level log? Not sure what norms we want to have around that kind of thing (just seems a bit nitty-gritty for info, if we print info level by default?)

File: gtfs_rt_validator_api.py
Comment (lauriemerrell) [inline gtfs_rt_validator_api.py:339]: Did we see any errors for other optional files? (Does this run successfully on *all* feeds with this line included?)

File: script/quick_test
Comment (lauriemerrell) [inline script/quick_test:1]: Can we add comments and/or docs about how to use this? (how to call it, what are the expected results?)

File: script/quick_test_many
Comment (lauriemerrell) [inline script/quick_test_many:1]: same comment as `quick_test`

File: script/seed_bucket_data.py
Comment (lauriemerrell) [inline script/seed_bucket_data.py:9]: Is this still needed/usable? Just wondering if it's worth adding a comment if it doesn't align any more (sorry I realize it's just getting picked up because of linter stuff, but given that it's here....)

**Review by lauriemerrell — COMMENTED** [review]
Overall looks good, a few mostly hygiene-type things.

File: gtfs_rt_validator_api.py
Comment (atvaccaro) [reply gtfs_rt_validator_api.py:164]: Oh yeah that's fair; I always just default to `info()` but I should definitely be better about that; fixing

File: gtfs_rt_validator_api.py
Comment (atvaccaro) [reply gtfs_rt_validator_api.py:339]: I haven't seen any so far; will see if I can execute on everything before I merge.

File: script/quick_test
Comment (atvaccaro) [reply script/quick_test:1]: yep can do

File: script/seed_bucket_data.py
Comment (atvaccaro) [reply script/seed_bucket_data.py:14]: Good question; I added a comment saying it was deprecated, for the same reason as the other thing

File: tests/test_download.py
Comment (lauriemerrell) [inline tests/test_download.py:19]: Just noticed this thing that still points to the `ci` directory.... Should we comment that this file is deprecated too?

File: tests/test_download.py
Comment (lauriemerrell) [inline tests/test_download.py:116]: Another hard coded `ci` reference

File: tests/test_download.py
Comment (lauriemerrell) [inline tests/test_download.py:141]: Another hard coded `ci` reference

File: tests/test_download.py
Comment (atvaccaro) [reply tests/test_download.py:19]: yeah.... honestly the tests shouldn't hit GCS at all

File: gtfs_rt_validator_api.py
Comment (atvaccaro) [reply gtfs_rt_validator_api.py:339]: I was able to execute a fair number and didn't hit a similar error; lots of missing RT data though

`finished multiprocessing; 208 successful of 651`

File: tests/test_download.py
Comment (atvaccaro) [reply tests/test_download.py:23]: I added a note that the tests need to be reworked. We probably need some pytest fixtures to help with this stuff.

File: gtfs_rt_validator_api.py
Comment (lauriemerrell) [reply gtfs_rt_validator_api.py]: looks like it's still `info` -- NBD but just checking if you had anything else you wanted to commit

**Review by lauriemerrell — APPROVED** [review]
LGTM, not merging myself because one open question on the logger levels, but I think it's good to go once that's confirmed.

File: gtfs_rt_validator_api.py
Comment (atvaccaro) [reply gtfs_rt_validator_api.py]: oops missed that, fixing

Comment (atvaccaro) [reply gtfs_rt_validator_api.py]: this one does say debug now

**Pull request merged**
